### PR TITLE
Match retail __osMalloc.c

### PR DESCRIPTION
**Pull request by cadmic**
Notably, retail has some new code to count the number of allocation failures, both per-arena and globally. This is probably a bug, because there's nothing that can read this info (the getter exists only in debug builds).

Some more notes:
* `NODE_IS_VALID`/`NODE_GET_NEXT`/`NODE_GET_PREV` are from Animal Forest (which is more similar to OOT than MM is here)
* `CHECK_ALLOC_FAILURE` was probably a debug macro, because the `do {} while (0)` helps match `__osRealloc` (otherwise we'd have to add an `if (1) {}`, h/t @inspectredc for matching this)
* I also created macros for using the arena debug flags (for filling/checking free/allocated blocks) because the same logic is used multiple times. I renamed the flags so I could use the verb forms for the macro names.
* I reordered the declarations in functions.h, because the debug functions are basically interleaved with the non-debug functions and I thought keeping them in order was too noisy.

Merge #1683 and #1700 first (removes usages of __osMalloc debug functions)

### CODE REVIEW

**Review by hensldm — APPROVED** [review]

File: include/z64.h [hunk 1]
```diff
@@ -715,12 +715,14 @@ typedef struct ArenaNode {
...
     /* 0x08 */ struct ArenaNode* next;
     /* 0x0C */ struct ArenaNode* prev;
+#if OOT_DEBUG
     /* 0x10 */ const char* filename;
     /* 0x14 */ s32 line;
     /* 0x18 */ OSId threadId;
     /* 0x1C */ Arena* arena;
     /* 0x20 */ OSTime time;
     /* 0x28 */ u8 unk_28[0x30-0x28]; // probably padding
+#endif
```
Comment (fig02) [inline include/z64.h:725]: worth noting that the n64 versions have all of this debug information in their heap nodes. obviously cant do anything about that right now, but this `#if` will change later

File: src/code/__osMalloc.c [hunk 2]
```diff
@@ -1,9 +1,9 @@
 #include "global.h"
 #include "terminal.h"
 
-#define FILL_ALLOCBLOCK (1 << 0)
-#define FILL_FREEBLOCK (1 << 1)
-#define CHECK_FREE_BLOCK (1 << 2)
+#define FLAG_FILL_ALLOC_BLOCK (1 << 0)
+#define FLAG_FILL_FREE_BLOCK (1 << 1)
+#define FLAG_CHECK_FREE_BLOCK (1 << 2)
```
[the remark this answers is not in the document]
Comment (cadmic) [reply src/code/__osMalloc.c:6]: Hm I wanted to use `FILL_ALLOC_BLOCK` as a function-like macro name. What do you suggest those macros should be called?

[the remark this answers is not in the document]
Comment (fig02) [reply src/code/__osMalloc.c:6]: oh I did not see that those were also the names of something else. Ill look it over again and see if I have any suggestions

File: include/z64.h (hunk 1 above)
Comment (cadmic) [reply include/z64.h]: I added a TODO

**Review by fig02 — APPROVED** [review]

**Pull request merged**